packages/vitest/src/integrations/coverage/istanbul.ts

```diff
@@ -0,0 +1,146 @@
+import { existsSync, promises as fs } from 'fs'
+import { createRequire } from 'module'
+import { resolve } from 'pathe'
+import type { ExistingRawSourceMap, TransformPluginContext } from 'rollup'
+
+import { configDefaults, defaultExclude, defaultInclude } from '../../defaults'
+import type { Vitest } from '../../node'
+import type { IstanbulOptions, ResolvedCoverageOptions } from '../../types'
+import type { BaseCoverageProvider } from './base'
+
+const require = createRequire(import.meta.url)
+const coverageVariable = '__VITEST_COVERAGE__'
+
+interface Instrumenter {
+  /* Instrument the supplied code and track coverage against the supplied filename. It throws if invalid code is passed to it. ES5 and ES6 syntax is supported. To instrument ES6 modules, make sure that you set the esModules property to true when creating the instrumenter. */
+  instrumentSync(
+    /* The code to instrument */
+    code: string,
+    /* The filename against which to track coverage. */
+    filename: string,
+    /* The source map that maps the not instrumented code back to it's original form. Is assigned to the coverage object and therefore, is available in the json output and can be used to remap the coverage to the untranspiled source.): string; */
+    inputSourceMap: object
+  ): string
+
+  /* Returns the file coverage object for the last file instrumented. */
+  lastSourceMap(): ExistingRawSourceMap
+}
+
+interface TestExclude {
+  new(opts: {
+    cwd?: string | string[]
+    include?: string | string[]
+    exclude?: string | string[]
+    extension?: string | string[]
+    excludeNodeModules?: boolean
+  }): { shouldInstrument(filePath: string): boolean }
+}
+
+export class IstanbulCoverageProvider implements BaseCoverageProvider {
+  ctx!: Vitest
+  options!: ResolvedCoverageOptions & { provider: 'istanbul' }
+  instrumenter!: Instrumenter
+  testExclude!: InstanceType<TestExclude>
+  coverages: any[] = []
+
+  initialize(ctx: Vitest) {
+    this.ctx = ctx
+    this.options = resolveIstanbulOptions(ctx.config.coverage, ctx.config.root)
+
+    const { createInstrumenter } = require('istanbul-lib-instrument')
```

Comment:
Would be cool to see this configurable to use custom instrumenters like https://github.com/kwonoj/swc-coverage-instrument/

Reply:
This seems like a cool idea, thanks for improvement idea @wight554. However I think it might be best if we leave this customization out of this PR and introduce it in a separate feature afterwards. 
Do you think there is anything important we need to take into consideration at this point? Leave some abstractions in place, etc?
My Rust knowledge is a bit _rusty_ and to me it's not that clear what exactly `import('swc-plugin-coverage-instrument')` would return here. Does it match the API of `istanbul-lib-instrument`?

Reply:
> This seems like a cool idea, thanks for improvement idea @wight554. However I think it might be best if we leave this customization out of this PR and introduce it in a separate feature afterwards. Do you think there is anything important we need to take into consideration at this point? Leave some abstractions in place, etc? My Rust knowledge is a bit _rusty_ and to me it's not that clear what exactly `import('swc-plugin-coverage-instrument')` would return here. Does it match the API of `istanbul-lib-instrument`?

I've checked how jest handles this, see https://github.com/facebook/jest/blob/24e0472ac41ea03cdd89ffaea8c5f410ecf6054f/packages/jest-transform/src/ScriptTransformer.ts#L410
In this case all that needs to be added is config value to skip instrumenter part and return unmodified map and code in `onFileTransform`